src/Shared/ProviderConfiguration.cs

```diff
@@ -117,6 +119,22 @@ private ProviderConfiguration(NameValueCollection config)
 
             ConnectionTimeoutInMilliSec = GetIntSettings(config, "connectionTimeoutInMilliseconds", 0);
             OperationTimeoutInMilliSec = GetIntSettings(config, "operationTimeoutInMilliseconds", 0);
+
+            var serializationTypeNameSpace = GetStringSettings(config, "sessionSerializationNamespaceAndType", null);
+            var serializationTypeAssembly = GetStringSettings(config, "sessionSerializationTypeAssembly", null);
+
+            if (!string.IsNullOrEmpty(serializationTypeNameSpace) && !string.IsNullOrEmpty(serializationTypeAssembly))
+            {
+                try
+                {
+                    var serializer = Activator.CreateInstance(serializationTypeAssembly, serializationTypeNameSpace);
```

Comment:
This is a very slow/dangerous/ugly way to instantiate things. Can callers pass their instance of `ISessionDataSerializer` to `ProviderConfiguration.SessionDataSerializer` instead of having us instantiate it?

Reply:
That's how we did it before: https://github.com/Azure/aspnet-redis-providers/blob/802a2c0721adc02002a7261519f86f551953f369/src/Shared/RedisUtility.cs#L22